deepinv/__main__.py

```diff
@@ -0,0 +1,284 @@
+# Usage: python -m deepinv train --config ./config.yaml
+
+from __future__ import annotations
+
+import deepinv as dinv
+import torch
+import torch.utils.data
+import torchvision.transforms as transforms
+
+import yaml
+from typing import Any
+from dataclasses import dataclass
+import argparse
+
+
+class CommandLineTrainer:
```

Comment:
For future extensibility, can we generalise this to just a `CLIParser` or something, that handles all the config ingestion, and then maybe have `CommandLineTrainer` as a subclass or something with only the core Trainer orchestration?